src/core/libraries/kernel/memory_management.cpp

```diff
@@ -207,6 +208,19 @@ int PS4_SYSV_ABI sceKernelQueryMemoryProtection(void* addr, void** start, void**
     return memory->QueryProtection(std::bit_cast<VAddr>(addr), start, end, prot);
 }
 
+int PS4_SYSV_ABI sceKernelMProtect(const void* addr, size_t size, int prot) {
+    Core::MemoryManager* memory_manager = Core::Memory::Instance();
+    Core::MemoryProt protection_flags = static_cast<Core::MemoryProt>(prot);
+    return memory_manager->Protect(std::bit_cast<VAddr>(addr), size, protection_flags);
+}
+
+int PS4_SYSV_ABI sceKernelMTypeProtect(const void* addr, size_t size, int mtype, int prot) {
```

Comment:
```suggestion
int32_t PS4_SYSV_ABI sceKernelMTypeProtect(const void* addr, size_t size, int mtype, int prot) {
```
Unfortunately I was unable to get the decompilation for the error calculation in this function...

Reply:
On my decomp setup, it seems like MTypeProtect calls a helper function that sets errno. Doesn't seem to be using any posix function though. Not sure what errno value it's using though. Then it returns the kernel equivalent to the errno value (or 0, if errno is 0).

Reply:
> On my decomp setup, it seems like MTypeProtect calls a helper function that sets errno. Doesn't seem to be using any posix function though. I'm also not sure what errno value it's using. Then it returns the kernel equivalent to the errno value (or 0, if errno is 0).

Perhaps you can post the integer value if there is one in this function? `2147------` or so, or convert this to hex.

Reply:
![Screenshot 2024-09-02 200405](https://github.com/user-attachments/assets/9982a0e9-1235-4dc8-bff2-b42952accc33)
This is what I was seeing at the time. If this is trustworthy, then the errno would be some 0x17b, or int 379, which makes no sense. Could be Ghidra bugging out though, as I've had a ton of weird issues with Ghidra when viewing libkernel in it.,

Reply:
Hmm this is at least similar to `sceKernelMProtect` in most ways, with usage of `__error`. If there is no extra information in the assembly, maybe can just copy error calculation same as `MProtect`, for now.